ui/desktop/app/routes/scopes/scope/authenticate.js

```diff
@@ -28,13 +29,42 @@ export default class ScopesScopeAuthenticateRoute extends Route {
    * scope and all scopes (for org navigation).
    * @return {Promise} `{scope, scopes, authMethods}`
    */
-  model() {
+  async model() {
     const { id: scope_id } = this.modelFor('scopes.scope');
 
+    // Fetch auth methods for current scope
+    const authMethods = await this.resourceFilterStore.queryBy(
+      'auth-method',
+      {
+        authorized_actions: [{ contains: 'authenticate' }],
+      },
+      {
+        scope_id,
+      },
+    );
+    // Preload all authenticatable auth methods into the store
+    const authMethodsForAllScopes = await this.resourceFilterStore.queryBy(
+      'auth-method',
+      {
+        authorized_actions: [{ contains: 'authenticate' }],
+      },
+      {
+        scope_id: 'global',
+        recursive: true,
+      },
+    );
+    // Fetch org scopes
+    // and filter out any that have no auth methods
+    const scopes = this.modelFor('scopes').filter(
+      ({ id: scope_id, isOrg }) =>
+        isOrg &&
+        authMethodsForAllScopes.filter((i) => i.scopeID === scope_id).length,
```

Comment:
We really shouldn't do a double `filter` here and instead create a set of valid scope IDs from `authMethodsForAllScopes` to do this check since we now know that some users can have a lot of scopes.

Reply:
Updated.